src/satosa/micro_services/attribute_generation.py

```diff
@@ -124,13 +125,13 @@ def __init__(self, config, *args, **kwargs):
     def _synthesize(self, attributes, requester, provider):
         syn_attributes = dict()
         context = dict()
-        
-        for attr_name,values in attributes.items():
-           context[attr_name] = MustachAttrValue(attr_name, values)
+
+        for attr_name, values in attributes.items():
+            context[attr_name] = MustachAttrValue(attr_name, values)
 
         recipes = get_dict_defaults(self.synthetic_attributes, requester, provider)
         for attr_name, fmt in recipes.items():
-           syn_attributes[attr_name] = [v.strip().strip(';') for v in re.split("[;\n]+", pystache.render(fmt, context))]
+            syn_attributes[attr_name] = list(filter(None, [v.strip().strip(';') for v in re.split("[;\n]+", pystache.render(fmt, context))]))
```

Comment:
It is not `None` that is filtered out, but empty strings. In general, Falsy values.

```suggestion
            syn_attributes[attr_name] = [
                value
                for token in re.split("[;\n]+", pystache.render(fmt, context))
                for value in [token.strip().strip(';')]
                if value
            ]
```